lib/sidekiq/web/application.rb

```diff
@@ -15,8 +15,8 @@ class WebApplication
       "manifest-src 'self'",
       "media-src 'self'",
       "object-src 'none'",
-      "script-src 'self' https: http:",
-      "style-src 'self' https: http: 'unsafe-inline'",
+      "script-src 'self' 'nonce-!placeholder!'",
+      "style-src 'self' 'nonce-!placeholder!'",
```

Comment:
I'm wondering if we can allow `style-src 'unsafe-inline'` until Sidekiq 8.0. Most people can extract JavaScript pretty easily but a lot of dynamic CSS isn't as easy to extract. Are you aware of any issues or trade offs here?

Reply:
`nonce` and `unsafe-inline` are not compatible with each other, when both are specified only `nonce` is respected, at least on modern browsers.

It seems fine to just undo the policy changes for `style-src` to allow for a smoother upgrade path, `script-src` is what this PR was all about. Loading arbitrary css is not exactly what you'd want to do but loading arbitrary scripts even less so.

Let me know what you think, I can make a PR for that. Or just do that yourself, either is fine for me.

Reply:
Ok, we'll continue to implement nonce but need to allow unsafe-inline for styles for now. We can enable nonce and disable unsafe-inline in 8.0. Extensions can use 7.3 to implement nonce so they can remain compatible with 7.3 and 8.0. I don't want to require a hard break in compatibility between major versions if possible.

Reply:
You're welcome to send a PR for that if you're in a good place to do so.

Reply:
Yeah, sure. I will do so. I'm currently looking into something else but will probably have time to do that tomorrow